script.lua

```diff
@@ -3229,57 +3229,97 @@ function setupPlayerArea(params)
 
     local energy = 0
 
-    local function elemStrToArr(elemStr)
-        local outArr = {}
-        for i = 1, string.len(elemStr) do
-            table.insert(outArr,(math.floor(string.sub(elemStr, i, i))))
-        end
-        return outArr
-    end
-
-    local function elemCombine(inTableOfElemStrCards)
+    local Elements = {}
+    Elements.__index = Elements
+    function Elements:new(init)
         local outTable = {0,0,0,0,0,0,0,0}
-        for i = 1, #inTableOfElemStrCards do
-            local elemTable = elemStrToArr(inTableOfElemStrCards[i].getVar("elements"))
+        setmetatable(outTable, self)
+        outTable:add(init)
+        return outTable
+    end
+    function Elements:add(other)
+        if other == nil then
+            return
+        elseif type(other) == "table" then
             for j = 1, 8 do
```

Comment:
the other conditional block uses i as a variable, for consistency this should also be i

Reply:
eh on second thought i can just make a couple formatting changes on my end after the fact